src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -277,15 +280,17 @@ private void setListMethodFields(
               typeTable,
               resourceField,
               resourceField.getType().makeOptional(),
-              resourceFieldConfig.getEntityName());
+              resourceFieldConfig.getEntityName(),
+              resourceNameType);
     } else {
       resourceTypeName = typeTable.getAndSaveNicknameForElementType(resourceField.getType());
     }
 
     String iterateMethodName =
         context
             .getNamer()
-            .getPagedResponseIterateMethod(context.getFeatureConfig(), resourceFieldConfig);
+            .getPagedResponseIterateMethod(
+                context.getFeatureConfig(), resourceFieldConfig, resourceNameType);
```

Comment:
I haven't thought enough about this to have an opinion, but would it make sense for resource name type to be an attribute of the field config?